app/main.py

```diff
@@ -1,34 +1,158 @@
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self, row: int, column: int) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = True
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(self, start: tuple, end: tuple) -> None:
+        self.start = start
+        self.end = end
+        self.is_drowned = False
+        self.decks = []
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+    def crate_decks(self, row: int, column: int) -> None:
+        self.decks.append(Deck(row, column))
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def get_deck(self, row: int, column: int) -> None:
+        for deck in self.decks:
+            if deck.row == row and deck.column == column:
+                return deck.is_alive
+
+    def fire(self, row: int, column: int) -> bool:
+        health_ship = []
+
+        for deck in self.decks:
+            if deck.row != row or deck.column != column:
+                health_ship.append(deck.is_alive)
+            elif deck.row == row and deck.column == column:
+                deck.is_alive = False
+                health_ship.append(deck.is_alive)
+        if not any(health_ship):
+            self.is_drowned = False
```

Comment:
Logic Error: The `is_drowned` attribute should be set to `True` when no decks are alive, indicating the ship is sunk. Currently, it is incorrectly set to `False`.